x-pack/plugins/ml/public/application/components/ml_inference/components/test_pipeline.tsx

```diff
@@ -0,0 +1,277 @@
+    try {
+      const pipelineConfig = getPipelineConfig(state);
+      const result = await simulateInferencePipeline(
+        JSON.parse(sampleDocsString) as IngestSimulateDocument[],
+        pipelineConfig
+      );
+      setSimulatePipelineResult(result);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      const errorProperties = extractErrorProperties(error);
+      setSimulatePipelineError(error);
+      toasts.danger({
+        title: i18n.translate(
+          'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.errorSimulatingPipeline',
+          {
+            defaultMessage: 'Unable to simulate pipeline.',
+          }
+        ),
+        body: errorProperties.message,
+        toastLifeTimeMs: 5000,
+      });
+    }
+  };
+
+  const clearResults = () => {
+    setSimulatePipelineResult(undefined);
+    setSimulatePipelineError(undefined);
+  };
+
+  const onChange = (json: string) => {
+    setSampleDocsString(json);
+    const valid = isValidJson(json);
+    setIsValid(valid);
+  };
+
+  const getSampleDocs = useCallback(async () => {
+    let records: IngestSimulateDocument[] = [];
+    let resp;
+
+    try {
+      resp = await esSearch({
+        index: sourceIndex,
+        body: {
+          size: 2,
+        },
+      });
+
+      if (resp && resp.hits.total.value > 0) {
+        records = resp.hits.hits;
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    }
+    setSampleDocsString(JSON.stringify(records, null, 2));
+  }, [sourceIndex, esSearch]);
+
+  useEffect(
+    function fetchSampleDocsFromSource() {
+      if (sourceIndex) {
+        getSampleDocs();
+      }
+    },
+    [sourceIndex, getSampleDocs]
+  );
+
+  return (
+    <>
+      <EuiFlexGroup direction="column" gutterSize="xs">
+        <EuiFlexItem>
+          <EuiTitle size="s">
+            <h4>
+              {i18n.translate(
+                'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.title',
+                { defaultMessage: 'Test your pipeline results' }
+              )}
+            </h4>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiText color="subdued" size="s">
+            <p>
+              <strong>
+                {i18n.translate(
+                  'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.optionalCallout',
+                  { defaultMessage: 'This is an optional step.' }
+                )}
+              </strong>
+              &nbsp;
+              <FormattedMessage
+                id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.description"
+                defaultMessage="Use this tool to run a simulation of your pipeline in order to confirm that it produces your anticipated results."
```

Comment:
```suggestion
                defaultMessage="Run a simulation of the pipeline to confirm it produces the anticipated results."
```

Reply:
Updated in 64ba3ac4ec279eb78f5f8a97d01005991ae76219